torch/csrc/autograd/variable.h

```diff
@@ -525,7 +567,8 @@ inline Variable make_variable(at::Tensor data, Edge gradient_edge) {
       !data.is_variable(),
       "Must not create a new variable from a variable, use its .data()");
   if (data.defined()) {
-    return Variable(c10::make_intrusive<Variable::Impl>(data, false, std::move(gradient_edge)));
+    auto data_copy = at::Tensor(data.getIntrusivePtr()->shallow_copy_and_detach());
+    return Variable(c10::make_intrusive<Variable::Impl>(data_copy, false, std::move(gradient_edge)));
```

Comment:
isn't this used in C++ a lot?  Not just from the python path?

Reply:
same question for the view function above.

Reply:
I believe both of them are used in C++ a lot, but the overhead should be small if the `shallow_copy_and_detach()` function is trivial enough.